ddtrace/contrib/aiopg/connection.py

```diff
@@ -82,6 +87,11 @@ def __init__(self, conn, pin=None, cursor_cls=AIOTracedCursor):
     def cursor(self, *args, **kwargs):
         # unfortunately we also need to patch this method as otherwise "self"
         # ends up being the aiopg connection object
+        if AIOPG_VERSION >= (0, 16, 0):
+            # Only one cursor per connection is allowed, as per DB API spec
+            self.close_cursor()
+            self._last_usage = self._loop.time()
```

Comment:
Not sure to understand why those lines are needed now and were not before. Is it possible to do the version check only at initialization/import time?

Reply:
Those two extra lines were added in 0.16.0 to check that there's only one cursor open per connection.

Reply:
Does aiopg>=0.16.0 always include 2 cursors per connection?

Reply:
> Does aiopg>=0.16.0 always include 2 cursors per connection?

The problem here is that a check was added on 0.16.0 to prevent opening a new cursor when an opened one already exists. So what the new method does is to close the existing one first and return a new one.